### PR TITLE
FIX react-native-test in CI

### PR DESCRIPTION
**Pull request by danielduan**
Issue:

react-native test is currently not running on master. it only logs a promise error and exits gracefully.

## What I did
(copy and fix from 3.3)

### CODE REVIEW

File: scripts/test.js [hunk 1]
```diff
@@ -66,7 +70,14 @@ const tasks = {
   //   defaultValue: false,
   //   option: '--reactnativeapp',
   //   projectLocation: './examples/crna-kitchen-sink',
+  //   isJest: true,
   // }),
+  cli: createProject({
```
Comment (Hypnosphi) [inline scripts/test.js]: We don't have CLI tests on `master`

**Review by Hypnosphi — CHANGES_REQUESTED** [review]

**Comment by ndelangen** [comment]
Should be ok now @Hypnosphi, thanks for taking notice 👍

**Review by ndelangen — APPROVED** [review]

File: scripts/test.js [hunk 2]
```diff
@@ -25,12 +25,13 @@ const spawn = command => {
...
 const main = program.version('3.0.0').option('--all', `Test everything ${chalk.gray('(all)')}`);
 
-const createProject = ({ defaultValue, option, name, projectLocation }) => ({
+const createProject = ({ defaultValue, option, name, projectLocation, isJest }) => ({
   value: false,
   defaultValue: defaultValue || false,
   option: option || undefined,
   name: name || 'unnamed task',
   projectLocation,
+  isJest,
```
Comment (Hypnosphi) [inline scripts/test.js]: This option can be removed as well (the only non-jest test is CLI)

**Review by Hypnosphi — CHANGES_REQUESTED** [review]

**Comment by Hypnosphi** [comment]
Please add some context to issue description, tell what this change is about

File: scripts/test.js [hunk 3]
```diff
@@ -156,14 +162,12 @@ if (
...
-      ' '
-    )}`;
-    console.log('command: ', command);
     if (list.length === 0) {
       log.warn(prefix, 'Nothing to test');
     } else {
-      spawn(command);
+      const projects = getProjects(list);
+      const extraParams = getExtraParams(list).join(' ');
+      spawn(`jest --projects ${jestProjects.join(' ')} ${extraParams}`);
```
Comment (Hypnosphi) [inline scripts/test.js]: `jestProjects` -> `projects.map(key => key.projectLocation)`

File: scripts/test.js [hunk 4]
```diff
@@ -85,19 +85,24 @@ const tasks = {
     option: '--runInBand',
     extraParam: '--runInBand',
   }),
+  update: createOption({
```
Comment (Hypnosphi) [inline scripts/test.js]: That's good that we're bringing this option to master, but let's backport the documentation as well then

**Review by Hypnosphi — CHANGES_REQUESTED** [review]

**Review by Hypnosphi — APPROVED** [review]

**Pull request merged**